### PR TITLE
GH-7 - map-entities

### LINKED ISSUE #7: Map entities with JPA specification
We need to prepare all the entities that were created on https://github.com/aleixoporpino/wealth-watch-api/pull/3, mapping and annotating with JPA/persistence specification.

### PR DESCRIPTION
**Pull request by luciannamend**
Mapped all entities and created Enum Currency. Closes #7

### CODE REVIEW

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java [hunk 1]
```diff
@@ -1,16 +1,58 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id",
+            columnDefinition = "INT",
+            nullable = false,
+            updatable = false)
     private Integer id;
+
+    @ManyToOne(targetEntity = User.class, cascade = CascadeType.ALL)
+    @JoinColumn(name = "user_id",
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: Lets keep these in one line

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java [hunk 2]
```diff
@@ -1,16 +1,58 @@
...
-    private Enum currency;
+
+    @Column(name = "name",
+            columnDefinition = "VARCHAR(50)",
+            nullable = false,
+            unique = true)
+        private String name;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "currency", nullable = false)
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: I think you missed the `columnDefinition` in this column

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java [hunk 3]
```diff
@@ -1,16 +1,58 @@
...
+    @Column(name = "created", nullable = false)
     private LocalDateTime created;
+
+    @UpdateTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
+    @Column(name = "updated", nullable = false)
     private LocalDateTime updated;
 
 }
+
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: Remove extra lines

File: src/main/java/com/porpapps/wealthwatchapi/model/User.java [hunk 4]
```diff
@@ -1,15 +1,56 @@
...
+    @Temporal(TemporalType.TIMESTAMP)
+    @Column(name = "created", nullable = false)
     private LocalDateTime created;
+
+    @UpdateTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
+    @Column(name = "updated", nullable = false)
     private LocalDateTime updated;
 }
+
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/model/User.java]: Remove extra lines

File: src/main/java/com/porpapps/wealthwatchapi/model/User.java [hunk 5]
```diff
@@ -1,15 +1,56 @@
...
+    @Column(name = "email",
+            columnDefinition = "VARCHAR(200)",
+            nullable = false,
+            unique = true)
     private String email;
+
+    @CreationTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
+    @Column(name = "created", nullable = false)
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/model/User.java]: Missing column definition for both created and updated

File: src/main/java/com/porpapps/wealthwatchapi/.idea/vcs.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/.idea/vcs.xml]: You need to delete these files, they shouldn't be here inside the package

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java [hunk 7]
```diff
@@ -1,16 +1,58 @@
...
+            unique = true)
+        private String name;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "currency", nullable = false)
+    private Currency currency;
+
+    @CreationTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
+    @Column(name = "created", nullable = false)
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: Missing column definition here as well.

**Comment by aleixoporpino** [comment]
Also, add a label for this PR, it should be a `task` label

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java (hunk 1 above)
Comment (luciannamend) [reply src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: done.

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java (hunk 2 above)
Comment (luciannamend) [reply src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: done.

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java (hunk 3 above)
Comment (luciannamend) [reply src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: done.

File: src/main/java/com/porpapps/wealthwatchapi/model/User.java (hunk 5 above)
Comment (luciannamend) [reply src/main/java/com/porpapps/wealthwatchapi/model/User.java]: done.

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java (hunk 7 above)
Comment (luciannamend) [reply src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java]: done.

File: src/main/java/com/porpapps/wealthwatchapi/.idea/vcs.xml (hunk 6 above)
Comment (luciannamend) [reply src/main/java/com/porpapps/wealthwatchapi/.idea/vcs.xml]: done.

File: src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java [hunk 8]
```diff
@@ -1,16 +1,48 @@
 package com.porpapps.wealthwatchapi.model;
 
+import jakarta.persistence.*;
 import lombok.Data;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
 
 import java.time.LocalDateTime;
 
 @Data
```
Comment (aleixoporpino) [inline src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java:12]: Also, add following annotation for each one of the models 
```java
@AllArgsConstructor
@NoArgsConstructor
```

Comment (luciannamend) [reply src/main/java/com/porpapps/wealthwatchapi/model/Portfolio.java:12]: done.

**Review by aleixoporpino — APPROVED** [review]

**Pull request merged**